stores/governance/src/main/java/com/bloxbean/cardano/yaci/store/governance/processor/LocalGovActionStateProcessor.java

```diff
@@ -0,0 +1,60 @@
+package com.bloxbean.cardano.yaci.store.governance.processor;
+
+import com.bloxbean.cardano.yaci.store.events.EpochChangeEvent;
+import com.bloxbean.cardano.yaci.store.events.RollbackEvent;
+import com.bloxbean.cardano.yaci.store.governance.service.LocalGovStateService;
+import com.bloxbean.cardano.yaci.store.governance.storage.impl.repository.*;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.context.event.EventListener;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.stereotype.Component;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.concurrent.TimeUnit;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+@ConditionalOnBean(LocalGovStateService.class)
+public class LocalGovActionStateProcessor {
+    private final LocalGovStateService localGovStateService;
+    private final LocalGovActionProposalStatusRepository localGovActionProposalStatusRepository;
+    private final LocalConstitutionRepository localConstitutionRepository;
+    private final LocalCommitteeMemberRepository localCommitteeMemberRepository;
+    private final LocalCommitteeRepository localCommitteeRepository;
+    private final LocalTreasuryWithdrawalRepository localTreasuryWithdrawalRepository;
+
+    @EventListener
+    @Transactional
+    public void handleEpochChangeEvent(EpochChangeEvent epochChangeEvent) {
+        if (!epochChangeEvent.getEventMetadata().isSyncMode())
+            return;
+
+        log.info("Epoch change event received. Fetching and updating local gov state");
+        localGovStateService.fetchAndSetGovState();
+    }
+
+    @EventListener
+    @Transactional
+    public void handleRollbackEvent(RollbackEvent rollbackEvent) {
+        long slot = rollbackEvent.getRollbackTo().getSlot();
+        int count = localGovActionProposalStatusRepository.deleteBySlotGreaterThan(slot);
+        localConstitutionRepository.deleteBySlotGreaterThan(slot);
+        localCommitteeRepository.deleteBySlotGreaterThan(slot);
+        localCommitteeMemberRepository.deleteBySlotGreaterThan(slot);
+        localTreasuryWithdrawalRepository.deleteBySlotGreaterThan(slot);
+
+        if (count > 0) {
+            log.info("Fetch gov state after rollback event...");
+            localGovStateService.fetchAndSetGovState();
+        }
+    }
+
+    @Scheduled(fixedRate = 2, timeUnit = TimeUnit.MINUTES)
+    public void scheduleFetchAndSetCurrentProtocolParams() {
```

Comment:
Is it going to handle the scenario mentioned in my previous comment ?

Does scheduler check if sync mode = true ? Also, do we need this scheduler?

Reply:
The reason I use the scheduler is that I have encountered a situation where, at the first block of an epoch, I performed a query to the node, but the result returned was similar to the previous epoch. After a few more blocks, I queried again, and the result returned had new information. Therefore, I need the scheduler to ensure accuracy.

However, I will recheck this to confirm whether the scheduler is really necessary or if there is a more reasonable alternative.

Reply:
Thanks @Sotatek-HuyLe3a 

Can you please verify it again after the recent slot # fix in era table? We had similar issue in local_epoch_param table. It was due to wrong epoch no. But now it's ok